src/openscm_runner/adapters/ciceroscm_adapter/ciceroscm.py

```diff
@@ -0,0 +1,133 @@
+
+
+def _ensure_dir_exists(path):
+    """
+    Ensure directory exists, and if it doesn't make it
+    """
+    if not os.path.exists(path):
+        os.mkdir(path)
+
+
+class CICEROSCM(_Adapter):  # pylint: disable=too-few-public-methods
+    """
+    Adapter for CICEROSCM
+    """
+
+    def __init__(self):
+        """
+        Initialise the CICEROSCM adapter
+
+        """
+        super().__init__()
+        self.odir = os.path.join(
+            os.path.dirname(__file__), "./utils_templates/openscm-test-datadir/"
+        )
+        self.udir = os.path.join(os.path.dirname(__file__), "./utils_templates/")
+        self.rundir = os.path.join(
+            os.path.dirname(__file__), "./utils_templates/run-dir/"
+        )
+        self.maindir = os.getcwd()
+        self.sfilewriter = SCENARIOFILEWRITER(self.udir)
+        self.pamfilewriter = PARAMETERFILEWRITER(self.udir)
+        self.resultsreader = CSCMREADER(self.odir)
+
+    def _init_model(self):  # pylint:disable=arguments-differ
+        pass
+
+    def _run(self, scenarios, cfgs, output_variables, output_config):
+        """
+        Run the model.
+
+        This method is the internal implementation of the :meth:`run` interface
+
+        cfgs is a list of indices to run
+        """
+        if output_config is not None:
+            raise NotImplementedError("`output_config` not implemented for Cicero-SCM")
+        first_time = True
+        # Loop over scenarios and write scenariofiles
+        for scen in scenarios.scenarios():
+            self.make_dir_structure(scen)
+            self.sfilewriter.write_scenario_data(
+                scenarios.filter(region="World").filter(scenario=scen),
+                "{odir}{scen}/inputfiles".format(odir=self.odir, scen=scen),
+            )
+            for pamset in cfgs:
+                self.pamfilewriter.write_parameterfile(
+                    pamset, "{odir}{scen}".format(odir=self.odir, scen=scen)
+                )
+                # Running:
+                os.chdir(self.rundir)
+                os.system(  # nosec # on Windows shell=True is required
+                    "./scm_vCH4fb ../openscm-test-datadir/{}/inputfiles/pam_current.scm".format(
+                        scen
+                    )
+                )
+                os.chdir(self.maindir)
+                for variable in output_variables:
+                    (
+                        years,
+                        timeseries,
+                        unit,
+                    ) = self.resultsreader.read_variable_timeseries(
+                        scen, variable, self.sfilewriter
+                    )
+                    if years.empty:
+                        continue
+                    if first_time:
+                        scm_df = ScmRun(
+                            pd.Series(timeseries, index=years),
+                            columns={
+                                "model": ["Cicero-SCM-{}".format(pamset["Index"])],
```

Comment:
Splitting the model and the parameter set into different metadata will make life much easier when handling the results
```suggestion
                                "model": "Cicero-SCM".
                                 "ensemble_member": pamset["Index"],
```